lib/parser/dockerfile/healthcheck.go

```diff
@@ -0,0 +1,157 @@
+//  Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerfile
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+// HeathcheckDirective represents the "LABEL" dockerfile command.
+type HealthcheckDirective struct {
+	*baseDirective
+
+	Interval    time.Duration
+	Timeout     time.Duration
+	StartPeriod time.Duration
+	Retries     int
+
+	Test []string
+}
+
+// Variables:
+//   Replaced from ARGs and ENVs from within our stage.
+// Formats:
+//   HEALTHCHECK NONE
+//   HEALTHCHECK [--interval=<t>] [--timeout=<t>] [--start-period=<t>] [--retries=<n>] \
+//     CMD ["<param>"...]
+//   HEALTHCHECK [--interval=<t>] [--timeout=<t>] [--start-period=<t>] [--retries=<n>] \
+//     CMD <command> <param>...
+func newHealthcheckDirective(base *baseDirective, state *parsingState) (Directive, error) {
+	// TODO: regexp is not the ideal solution.
+	if isNone := regexp.MustCompile(`^[\s|\\]*none[\s|\\]*$`).MatchString(base.Args); isNone {
+		return &HealthcheckDirective{
+			baseDirective: base,
+			Test:          []string{"None"},
+		}, nil
+	}
+	cmdIndices := regexp.MustCompile(`[\s|\\]*cmd[\s|\\]*`).FindStringIndex(base.Args)
+	if len(cmdIndices) < 2 {
+		return nil, base.err(fmt.Errorf("CMD not defined"))
+	}
+
+	flags, err := splitArgs(base.Args[:cmdIndices[0]])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse interval")
+	}
+
+	var interval, timeout, startPeriod time.Duration
```

Comment:
Does docker not have defaults for these? Is it just 0?

Reply:
My understanding is docker daemon will handle 0. Need more test for this.

Reply:
Seems to work.